### DISCUSSION TITLE
How to change output URL/permalink 'pattern' for all pages?

### DISCUSSION THREAD

**Discussion by kai687** [11ty/eleventy#1667] [Q&A] [answered]
Is it possible to remove certain parts of the permalink 'path' for all pages? 

For example, all my Markdown files are in a `content/` folder in the input directory. 
Let's say, I have a page `./content/about.md`. This page will be available at `/content/about/`. What I want is `/about/`. 

I know how to change the permalink for single pages, but I want to change that for every page inside the `./content` directory, always removing the `/content` part from the URL. 

Naïvely, I tried using a data file `content.json`, and place `permalink: {{ page.url | replace('/content', '') | url }}` inside. I get errors about duplicate permalinks. 

If I use a `js` data file: 

```js
module.exports = {
  layout: "markdown",
  eleventyComputed: {
    permalink: (data) => data.page.url.replace("/content", ""),
  },
};
```

I get a `Error: Dependency Cycle Found: page.url -> permalink -> page.url` error. 

Is that something that should work and I am doing something wrong, or is that currently not supported?

**Comment by kai687** [comment, marked as answer]
Ok, well: `{{ page.filePathStem | replace('/content', '') | url }}` seemed to have done the trick. Sorry for the spam.

**Reply by pdehaan** [reply]
I think I had to append a trailing `/` to the `filePathStem` locally to avoid having it create a "/contact" file vs a "/contact/index.html".

```js
permalink: "{{ page.filePathStem | replace: '/content', '' | append: '/' | url }}"
```

And if I remember correctly, you might run into problems if you have a /content/index.md file since it will try creating a "/index/index.html" by default. 

```js
{
  "date": "2021-02-28T19:30:25.768Z",
  "inputPath": "./src/content/index.md",
  "fileSlug": "content",
  "filePathStem": "/content/index",
  "url": "/index/",
  "outputPath": "www/index/index.html"
}
```
